java/server/test/org/openqa/selenium/grid/distributor/local/LocalDistributorTest.java

```diff
@@ -0,0 +1,82 @@
+package org.openqa.selenium.grid.distributor.local;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.ImmutableCapabilities;
+import org.openqa.selenium.events.EventBus;
+import org.openqa.selenium.events.local.GuavaEventBus;
+import org.openqa.selenium.grid.data.DistributorStatus;
+import org.openqa.selenium.grid.data.Session;
+import org.openqa.selenium.grid.distributor.Distributor;
+import org.openqa.selenium.grid.distributor.local.LocalDistributor;
+import org.openqa.selenium.grid.node.Node;
+import org.openqa.selenium.grid.node.local.LocalNode;
+import org.openqa.selenium.grid.sessionmap.SessionMap;
+import org.openqa.selenium.grid.sessionmap.local.LocalSessionMap;
+import org.openqa.selenium.grid.testing.TestSessionFactory;
+import org.openqa.selenium.remote.SessionId;
+import org.openqa.selenium.remote.http.HttpClient;
+import org.openqa.selenium.remote.http.HttpHandler;
+import org.openqa.selenium.remote.http.HttpRequest;
+import org.openqa.selenium.remote.http.HttpResponse;
+import org.openqa.selenium.remote.tracing.DistributedTracer;
+
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Set;
+import java.util.UUID;
+
+public class LocalDistributorTest {
+  private DistributedTracer tracer;
+  private EventBus bus;
+  private HttpClient.Factory clientFactory;
+  private URI uri;
+  private Node local;
+
+  @Before
+  public void setUp() throws URISyntaxException {
+    tracer = DistributedTracer.builder().build();
+    bus = new GuavaEventBus();
+    clientFactory = HttpClient.Factory.createDefault();
+
+    //Borrowed from the NodeTest code
+    class Handler extends Session implements HttpHandler {
+      private Handler(Capabilities capabilities) {
+        super(new SessionId(UUID.randomUUID()), uri, capabilities);
+      }
+
+      @Override
+      public HttpResponse execute(HttpRequest req) throws UncheckedIOException {
+        return new HttpResponse();
+      }
+    }
+
+    ImmutableCapabilities caps = new ImmutableCapabilities("browserName", "cheese");
```

Comment:
Prefer using "the most general interface or class that captures intent properly" on the left hand side (which is a long way of saying `Capabilities` ;) )

Reply:
Lol, that's one of my favorite pet issues... can't believe I missed one!

Reply:
I believe I've fixed this in the revised PR